site/en/docs/extensions/mv3/service_workers/index.md

```diff
@@ -108,22 +113,15 @@ chrome.runtime.onInstalled.addListener(() => {
   });
 });
 ```
+{% endCompare %}
 
-Extensions can remove listeners from their background scripts by calling `removeListener`. If all
-listeners for an event are removed, Chrome will no longer load the extension's background script for
-that event.
-
-```js
-chrome.runtime.onMessage.addListener((message, sender, reply) => {
-  chrome.runtime.onMessage.removeListener(event);
-});
-```
+Extensions can remove listeners from their background scripts by calling `removeListener()`. 
 
 ## Filter events {: #filters }
 
-Use APIs that support [event filters][8] to restrict listeners to the cases the extension cares
-about. If an extension is listening for the [`tabs.onUpdated`][9] event, try using the
-[`webNavigation.onCompleted`][10] event with filters instead, as the tabs API does not support
+Use APIs that support [event filters][8] to restrict listeners to specific use cases. If an
+extension is listening for the [`tabs.onUpdated`][9] event, try using the
+[`webNavigation.onCompleted`][10] event with filters instead, as the `onUpdated` event does not support
 filters.
```

Comment:
tabs.onUpdated` and `webNavigation.onCompleted` are aimed at different uses, so this suggestion felt weird to me. As is my way, I inserted more context on this scenario to make it read better to me. I defer to you on whether this is a good change.

```suggestion
Use APIs that support [event filters][8] to restrict listeners to specific use cases. If an
extension is listening for the [`tabs.onUpdated`][9] event to detect when a user navigates
to a specific website, try using the [`webNavigation.onCompleted`][10] event with filters
instead, as the `tabs.onUpdated` event does not support filters.
```